### PR TITLE
Initial/Final state writing to LHE files

### PR DESCRIPTION
**Pull request by marcorossi5**
This should address #1 

The LheWriter now gets the initial state pids directly from the ME

We are missing the final state, that could be included as an attribute into the ME.  Currently running madflow with the --histograms flag supports `t t~` final state only.

### CODE REVIEW

**Review by marcorossi5 — COMMENTED** [review]
I've the feeling that something is wrong  here. Please, let me know if you agree

File: python_package/madflow/lhe_writer.py [hunk 1]
```diff
@@ -157,57 +200,61 @@ def lhe_parser(self, all_ps, res):
...
+                        "pz": ps[3],
+                        "mass": np.sqrt(
+                            ps[0] ** 2 - ps[1] ** 2 - ps[2] ** 2 - ps[3] ** 2
+                        ),  # vectorize this?
+                        "vtim": 0,
+                        "helicity": 0,
+                    }
+                    for pid, status, ps in zip(pids, status_vec, ps_external)
+                ]
+            )
```
Comment (scarlehoff) [inline python_package/madflow/lhe_writer.py:252]: If you move these loops to `self.dump_events` (which is call asynchronously) and then just pass the information you need, you get a huge speed up as the dumping is performed while the next iteration is running (so effectively only the last one will maybe have to wait for a little longer).

**Comment by scarlehoff** [comment]
What is the state of this? Can we now (once conflicts are fixed) use the LHE parser with any process?

(maybe we talked about this before the holidays but I didn't note anything down, so it would be all in RAM and I went on maintenance shutdown during the summer :P)

**Comment by marcorossi5** [comment]
I @scarlehoff, I think that initial state is almost done, while we have something to discuss about final states.

I am on holidays until next week. We can then resume the development!

**Pull request closed without merging**